clinica/converters/_utils.py

```diff
@@ -588,3 +630,48 @@ def get_subjects_list_from_file(subjects_list_path: Path) -> list[str]:
         List of subjects.
     """
     return subjects_list_path.read_text().splitlines()
+
+
+def get_subject_id_pattern(study_name: StudyName) -> str:
```

Comment:
We can delete this function and rely on `BIDSSubjectID` (see `clinica.converters.study_model`) which already has the logic to convert IDs from their raw values to BIDS values and vice versa. 
Sorry for the bad suggestion, I hadn't thought about it hard enough...